support/cas-server-support-discovery-profile/src/main/java/org/apereo/cas/discovery/CasServerProfileRegistrar.java

```diff
@@ -122,6 +124,25 @@ private Object locateSubtypesByReflection(final Function<Class, Object> mapper,
         return clients.findAllClients().stream().map(Client::getName).collect(Collectors.toSet());
     }
 
+    private Set<String> locateAvailableAttributes() {
+        final LinkedHashSet<String> attributes = new LinkedHashSet<>(0);
+        if (attributeRepository != null) {
+            attributes.addAll(attributeRepository.getPossibleUserAttributeNames());
+        }
+        if (casProperties.getAuthn().getLdap() != null) {
+            casProperties.getAuthn().getLdap().stream()
+                    .forEach(ldap -> {
+                        attributes.addAll(ldap.getPrincipalAttributeList());
```

Comment:
This will not work. The list may have attributes that are mapped to something else. You cant take the Map as is and you need to transform it. See examples in the codebase on how the list is processed. The same applies to the JDBC stuff below.